eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -5,23 +5,23 @@
   },
   "matrix": {
     "Agent": {
-      "ubuntu-18.04": { "OSVmImage": "MMSUbuntu18.04", "Pool": "azsdk-pool-mms-ubuntu-1804-general" },
-      "windows-2019": { "OSVmImage": "MMS2019", "Pool": "azsdk-pool-mms-win-2019-general" },
+      "ubuntu-18.04": { "OSVmImage": "MMSUbuntu18.04", "Pool": "azsdk-pool-mms-ubuntu-1804-storage" },
```

Comment:
cc @benbp 

@mitchdenny what strategy are you thinking about for overriding the pools we use here? Should we enable some way to override these values in the template instead of requiring that folks duplicate the entire matrix.json file?

Reply:
@weshaggard when I was trying to show the pool override alternative, I found a small bug that made it not work :)

With the current import + override behavior (predicated on the above bug fix), this is what it could look like:

```
{
  "matrix": {
    "$IMPORT": "./eng/pipelines/templates/stages/platform-matrix.json",
    "PoolOverride": {
      "StoragePoolLinux": { "Pool": "azsdk-pool-mms-ubuntu-1804-storage" },
      "StoragePoolWin": { "Pool": "azsdk-pool-mms-win-2019-storage" },
      "HostedPoolMac": { "Pool": "Azure Pipelines" }
    }
  },
  "exclude": [
    {
      "Pool": "StoragePoolLinux",
      "OSVmImage": [ "MMS2019", "MacOS-10.15" ]
    },
    {
      "Pool": "StoragePoolWin",
      "OSVmImage": [ "MMSUbuntu18.04", "MacOS-10.15" ]
    },
    {
      "Pool": "HostedPoolMac",
      "OSVmImage": [ "MMS2019", "MMSUbuntu18.04" ]
    }
  ]
}
```

It might be worth thinking about a `replace` type operator in the config for this kind of scenario.

Reply:
Fix for the bug I mentioned: https://github.com/Azure/azure-sdk-tools/pull/1461

Reply:
Yeah it might be interesting to have some sort of replace operator that could update the entire agent based on the name of the agent.

Reply:
Let me sketch something out.